### PR TITLE
[state store ttl] Add context to persisting state

### LINKED ISSUE #3415: State TTL -1 is missing some context
This description is a little thin / ambiguous : 

image

https://docs.dapr.io/developing-applications/building-blocks/state-management/state-store-ttl/#persisting-state-ignoring-an-existing-ttl

bernd provdes the following additional context in [discord](https://discord.com/channels/778680217417809931/901141713089863710/1107811284025626825) which would be nice to include in the description :

> It means - sometimes you might have a TTL set at the DB / container level, and many databases allow you to overwrite this for a specific entity. So you can tell it to explicitly PERSIST the given key. No TTL specified at the entity level would mean the global TTL is applied, but specifying -1 means to explicitly persist it.

### PR DESCRIPTION
**Pull request by hhunter-ms**
## Description

Add context provided by Bernd in Discord to the state store ttl 'persist state' section.

## Issue reference

PR will close: #3415

### CODE REVIEW

**Comment by msfussell** [comment]
@olitomlinson

**Comment by olitomlinson** [comment]
@msfussell Thanks!

I'm not sure 'explicitly persist' is the right words to describe whats happening here. 

_Persisting_ of state is a given, right?

This is about hinting to the underlying component to not apply any global TTLs.

**note :** Also, if this `-1` semantic is only applicable to CosmoDb, does it make sense that this is in the docs?

**note :** This probably isn't the time or place, but `-1` isn't a great way of exposing this capability. It feels very arbitrary. I would prefer something more high level i.e. pass a string `IGNORE_TTL` or something like that at the Dapr API level, which is then translated into component specific instructions i.e. `-1` for cosmosDb etc.

**Comment by hhunter-ms** [comment]
@olitomlinson  thanks for your review! 

> 
> **note :** Also, if this `-1` semantic is only applicable to CosmoDb, does it make sense that this is in the docs?
> 

I can just move this to the CosmosDB component spec, if that works?

**Comment by msfussell** [comment]
@olitomlinson - Is this really only for CosmosDB? The way @berndverst  wrote this I thought he meant all. @berndverst ?

**Comment by berndverst** [comment]
This applies to all state stores which either at the Dapr component level - or when creating the resource outside of Dapr - let you specify a default TTL that is used for all data.
In those cases when no TTL is specified the data would expire after that period of time - which is not facilitated by Dapr at all.

Now all of such state stores also support the option to explicitly persist data anyway, which means to ignore the default database policy (which may or may not have been set up using Dapr) to indefinitely retain a given database record. This is done by setting "ttlInSeconds" to the value "-1"

Hope this is clear.

"Explicit persistence bypassing globally defined TTL" or something along those lines is the way to describe this.

File: daprdocs/content/en/developing-applications/building-blocks/state-management/state-store-ttl.md [hunk 1]
```diff
@@ -16,9 +16,15 @@ When state TTL has native support in the state store component, Dapr forwards th
 
 When a TTL is not specified, the default behavior of the state store is retained.
 
-## Persisting state (ignoring an existing TTL)
+## Explicit persistence bypassing globally defined TTL
 
-To explicitly persist a state (ignoring any TTLs set for the key), specify a `ttlInSeconds` value of `-1`.
+Persisting state applies to all state stores that let you specify a default TTL used for all data, either:
+- At the Dapr component level, or 
+- When creating the resource outside of Dapr.
```
Comment (msfussell) [inline daprdocs/content/en/developing-applications/building-blocks/state-management/state-store-ttl.md]:
```suggestion
- When creating the state store outside of Dapr and setting a global TTL value. 
```

File: daprdocs/content/en/developing-applications/building-blocks/state-management/state-store-ttl.md [hunk 2]
```diff
@@ -16,9 +16,15 @@ When state TTL has native support in the state store component, Dapr forwards th
 
 When a TTL is not specified, the default behavior of the state store is retained.
 
-## Persisting state (ignoring an existing TTL)
+## Explicit persistence bypassing globally defined TTL
 
-To explicitly persist a state (ignoring any TTLs set for the key), specify a `ttlInSeconds` value of `-1`.
+Persisting state applies to all state stores that let you specify a default TTL used for all data, either:
+- At the Dapr component level, or
```
Comment (msfussell) [inline daprdocs/content/en/developing-applications/building-blocks/state-management/state-store-ttl.md]:
```suggestion
- Setting a global TTL value via a Dapr component, or 
```

File: daprdocs/content/en/developing-applications/building-blocks/state-management/state-store-ttl.md [hunk 3]
```diff
@@ -16,9 +16,15 @@ When state TTL has native support in the state store component, Dapr forwards th
...
 
-## Persisting state (ignoring an existing TTL)
+## Explicit persistence bypassing globally defined TTL
 
-To explicitly persist a state (ignoring any TTLs set for the key), specify a `ttlInSeconds` value of `-1`.
+Persisting state applies to all state stores that let you specify a default TTL used for all data, either:
+- At the Dapr component level, or 
+- When creating the resource outside of Dapr. 
+
+When no TTL is specified, the data expires after that default period of time. This is not facilitated by Dapr at all.
```
Comment (msfussell) [inline daprdocs/content/en/developing-applications/building-blocks/state-management/state-store-ttl.md]:
```suggestion
When no specific TTL is specified, the data expires after that global TTL period of time. This is not facilitated by Dapr.
```

File: daprdocs/content/en/developing-applications/building-blocks/state-management/state-store-ttl.md [hunk 4]
```diff
@@ -16,9 +16,15 @@ When state TTL has native support in the state store component, Dapr forwards th
...
+
+In addition, all of these state stores also support the option to _explicitly_ persist data. This means they ignore the default database policy (which may or may not have been set up using Dapr) to indefinitely retain a given database record. You can do this by setting `ttlInSeconds` to the value `-1`.
```
Comment (msfussell) [inline daprdocs/content/en/developing-applications/building-blocks/state-management/state-store-ttl.md]:
```suggestion
In addition, all state stores also support the option to _explicitly_ persist data. This means you can ignore the default database policy (which may have been set outside of Dapr or via a Dapr Component) to indefinitely retain a given database record. You can do this by setting `ttlInSeconds` to the value of `-1`. This value indicates to ignore any TTL value set.
```

**Review by msfussell — COMMENTED** [review]
Here are my suggestions

**Review by msfussell — APPROVED** [review]
LGTM

**Pull request merged**